src/Catalyst.Core.Modules.Web3/Web3EthApiExtensions.cs

```diff
@@ -0,0 +1,194 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using Catalyst.Abstractions.Kvm.Models;
+using Catalyst.Abstractions.Ledger;
+using Catalyst.Abstractions.Repository;
+using Catalyst.Core.Lib.Extensions;
+using Catalyst.Core.Modules.Web3.Controllers.Handlers;
+using Catalyst.Protocol.Deltas;
+using Catalyst.Protocol.Transaction;
+using Google.Protobuf;
+using Google.Protobuf.WellKnownTypes;
+using LibP2P;
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Dirichlet.Numerics;
+using Nethermind.Evm.Tracing;
+
+namespace Catalyst.Core.Modules.Web3 
+{
+    public static class Web3EthApiExtensions
+    {
+        public static DeltaWithCid GetLatestDeltaWithCid(this IWeb3EthApi api)
+        {
+            return GetDeltaWithCid(api, api.DeltaResolver.LatestDelta);
+        }
+
+        public static DeltaWithCid GetDeltaWithCid(this IWeb3EthApi api, Cid cid)
+        {
+            if (!api.DeltaCache.TryGetOrAddConfirmedDelta(cid, out Delta delta))
+            {
+                throw new Exception($"Delta not found '{cid}'");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['System.Exception' should not be thrown by user code.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4585091)